Fix Name Wrapper interface issues

nxt3d opened the pull request:
This PR fixes a number of issues with the INameWrapper interface including:
- Removing a duplicate INameWrapper interface
- Adding missing functions to the INameWrapper, e.g. isWrapped
- Removing unnecessary interface override statements. 

In order to get the tests to pass I needed to clear old build data. I used:
```
rm -fr node_modules build cache artifacts
yarn
```
@jefflau

jefflau commented:
I've been thinking about `isWrapped()`, and I think we should probably make at least this version of it internal, since it does not check for .eth names.

Also note upgrade contract has been changed (and just merged) due to an issue we found, so this would change some of those function signatures.

> Removing unnecessary interface override statements.

I thought override/virtual was required on functions that are set in the interface and then overwritten?

nxt3d commented:
>I've been thinking about isWrapped(), and I think we should probably make at least this version of it internal, since it does not check for .eth names. Possibly we could expose an offchain version that does all the checks and does not have to be gas efficient

ok

> Also note upgrade contract has been changed (and just merged) due to an issue we found, so this would change some of those function signatures.

noted

> I thought override/virtual was required on functions that are set in the interface and then overwritten?

It's not necessary since 0.8.8
https://docs.soliditylang.org/en/v0.8.18/contracts.html#function-overriding

>Also, can we separate out the Resolver.sol interfaces, so we can look at them separately instead of bundled with NameWrapper interfaces

ok

nxt3d commented:
I rebased my changes on the latest master branch, updated isWrapped to be internal, and took out the Resolver.sol interface updates. I will push that as a separate PR. @jefflau

jefflau commented:
@nxt3d Can you add `upgrade()` to the interface as well? https://github.com/ensdomains/ens-contracts/blob/master/contracts/wrapper/NameWrapper.sol#L481

nxt3d commented:
> @nxt3d Can you add `upgrade()` to the interface as well?

When I rebased, upgrade was added to the INameWrapper, so it's there now.

jefflau commented:
Can we move `isWrapped()` to the internal section of the contracts and prefix it with an underscore. I don't think we need the comment since it's no longer public

jefflau commented:
I've merged and I'll make the updates in a separate branch
